model/task_queue_service_dependency.go

```diff
@@ -239,8 +238,6 @@ func (d *basicCachedDAGDispatcherImpl) rebuild(items []TaskQueueItem) error {
 
 // FindNextTask returns the next dispatchable task in the queue, and returns the tasks that need to be checked for dependencies.
 func (d *basicCachedDAGDispatcherImpl) FindNextTask(ctx context.Context, spec TaskSpec, amiUpdatedTime time.Time) *TaskQueueItem {
-	d.mu.Lock()
```

Comment:
I notice we remove this here and add it into the task group function; do we need to have it anywhere that isn't using task groups? getNodeByItemID or something? (Also, could we put a comment on this function about why we don't lock, in case we think it's not safe enough later?)